charts/rasa-x/values.yaml

```diff
@@ -279,8 +279,9 @@ images:
 
 # securityContext to use
 securityContext:
-#  # fsGroup to use (e.g. when deploying on Kubernetes)
-#  fsGroup: 0
+  runAsUser: 1000
```

Comment:
<img width="1728" alt="Screenshot 2020-02-07 at 11 06 45" src="https://user-images.githubusercontent.com/7667273/74020490-f48e1e80-4999-11ea-941d-433823d841e6.png">

This doesn't work in OpenShift. How did you test it in https://github.com/RasaHQ/rasa-x-helm/issues/11#issuecomment-581299375 ?

Reply:
What cluster did you use? Because it can also be because of the security settings of your cluster.
Here is the relevant doc: https://access.redhat.com/documentation/en-us/openshift_container_platform/3.11/html/cluster_administration/admin-guide-manage-scc

I tried it with Red Hat CodeReady Containers (OpenShift 4.x) and it worked well.

Reply:
This might also be useful: https://github.com/rook/rook/issues/1314

Reply:
I did the same thing on a plain crc install 🙈 Did you also do the `crc start` / `crc stop` thing? Downloaded it from here https://developers.redhat.com/products/codeready-containers . I am also pretty sure that these are the default settings. 

And I don't think we can do the `RunAsAny` because that's basically the thing about OpenShift that things are not able to run as `root`. I don't think our customers will addapt their security context for our Helm chart, what do you think?

Reply:
`crc version` gives me this:
```
crc version: 1.2.0+c2e3c0f
OpenShift version: 4.2.8 (embedded in binary)
```

Reply:
Mine is different. Have no idea why because I installed it using exactly the same URL :)

`>  ~ crc version
crc version: 1.5.0+e1c8fb8
OpenShift version: 4.2.14 (embedded in binary)`

Reply:
mhm, maybe they changed sth in one of the last versions? Our own OpenShift cluster is still on 3.11.59

Reply:
I'll download the newer crc version and try with that

Reply:
I wonder how OpenShift users are normally managed — do people actually adapt and specify security context manually? (because it seems to be a big thing in OpenShift — there is even a huge doc describing how to do that: https://docs.openshift.com/container-platform/4.1/authentication/managing-security-context-constraints.html).

Reply:
Alternatively we could also try the init container approach: 
https://github.com/helm/charts/blob/a09366729eb1ecad98d70df81030a28472cadd1a/stable/rabbitmq/templates/statefulset.yaml#L64

Reply:
> users are normally managed

OpenShift by default will always use a random user. The only constant is the group which is always `0` (root, but unlike the user the `root` group does not have special privileges)

Reply:
We can also just add a note for OpenShift users and explain that you have to override the default securityContext and replace it with something like `helm install rasa/rasa-x --set securityContext.fsGroup=""`

(according to the solutions people share here: https://github.com/bitnami/charts/issues/932)

Reply:
Just tried it out with the latest crc setup and that works. I'll include a note for OpenShift in my docs revamp then. Did you have any luck exploring the initContainer approach?

Reply:
The problem with this approach is that we cannot just execute something like `chown -R 1234:1234 /app` because we don't know the IDs in advance (OpenShift generates these IDs automatically). 
I also tried replacing `securityContext` with `{}` and adding this to `initContainers`
```
      initContainers:
      - name: volume-permissions
        image: alpine:3.6
        command: ["/bin/chown", "-R", "0777", "/app"]
        securityContext:
          runAsUser: 0
        volumeMounts:
        - name: rasa-x-claim
          mountPath: /app
```

It works but it's more like a hack and not a proper solution. I think we should actually add an OpenShift note to docs and revisit this issue after we stop running docker images as root.

Reply:
good for me! :-)